File: Managed/UnrealSharp/UnrealSharp/Interop/FQuatExporter.cs

using UnrealSharp.Binds;
using UnrealSharp.CoreUObject;

namespace UnrealSharp.Interop;

[NativeCallbacks]
public static unsafe partial class FQuatExporter
{
    public static delegate* unmanaged<out FQuat, FRotator, void> ToQuaternion;
    public static delegate* unmanaged<out FRotator, FQuat, void> ToRotator;
}

File: Source/UnrealSharpCore/Export/FQuatExporter.cpp
#include "FQuatExporter.h"

void UFQuatExporter::ToQuaternion(FQuat& Quaternion, const FRotator& Rotator)
{
	Quaternion = Rotator.Quaternion();
}

void UFQuatExporter::ToRotator(FRotator& Rotator, const FQuat& Quaternion)
{
	Rotator = Quaternion.Rotator();
}

